src/compiler/checker.ts

```diff
@@ -7515,6 +7527,54 @@ module ts {
             return flags & flagsToCheck;
         }
 
+        function checkDuplicateDeclarationOfSymbolInSameBlock(symbol: Symbol): void {
+            if ( !symbol.declarations) return;

```

Comment:
I think what you want to do is check whether or not there's more than 1 declaration. If there is only 1, bail out.